app/channel.go

```diff
@@ -1130,6 +1130,48 @@ func (a *App) GetChannelMembersPage(channelId string, page, perPage int) (*model
 	}
 }
 
+func (a *App) GetChannelMembersTimezones(channelId string) (model.ChannelTimezones, *model.AppError) {
+	var membersTimezones model.ChannelTimezones
+	if result := <-a.Srv.Store.Channel().GetChannelMembersTimezones(channelId); result.Err != nil {
+		return nil, result.Err
+	} else {
+		membersTimezones = result.Data.(model.ChannelTimezones)
+	}
+
+	// Remove timezones that was not set in manualTimezone or automaticTimezone
+	// also remove duplicate timezones
+	var newMemberTimezones model.ChannelTimezones
+	for i := range membersTimezones {
+		if membersTimezones[i].Timezone["manualTimezone"] == "" && membersTimezones[i].Timezone["automaticTimezone"] == "" {
```

Comment:
Is there a way to use something equivalent to `user.GetPreferredTimezone` here? This is ignoring the `useAutomaticTimezone` setting, and checking both manual/automatic timezones here is confusing.

Maybe you could do something to like
```go
// model/user.go
type User struct {
    // ...
    Timezone UserTimezone
}

func (u *User) GetPreferredTimezone() string {
    return u.Timezone.GetPreffered()
}

// model/user_timezone.go
type UserTimezone map[string]string

func (t *UserTimezone) GetPreferred() string {
    // copy what User.GetPreferredTimezone used to do
}

// app/channel.go
func (a *App) GetChannelMembersTimezones(channelId string) ([]model.UserTimezone, *model.AppError) {
    memberTimezones := ...

    timezones := []string{}
    for _, memberTimezone := range memberTimezones {
        timezones = append(timezones, memberTimezone.getPreferred())
    }

    return model.RemoveDuplicateStrings(timezones)
}
```

Reply:
1.  Since the `useAutomaticTimezone` value isn't being used, it incorrectly prioritizes automatically set timezones instead of manually set ones
2. For the return type, the way that duplicates are handled is kind of weird though since it checks for an existing one with either an automatic or manual timezone matching this one. Also, returning the whole timezone object for different users instead of just the actual timezones returns a lot of weird data. Like running this on pre-release, if it returns mine, it would return a bunch of weird, extra information (see below). I'd expect this API to just return a list of simple timezone strings
```
{
    "useAutomaticTimezone": "true",
    "automaticTimezone": "Americas/Toronto",
    "manualTimezone": "Asia/Seoul"
}
```

Reply:
this is weird, in my local server if you select manual time zone the automatictimezone is blank and the other way 
in my case:
```
{"automaticTimezone":"","manualTimezone":"Africa/Asmera","useAutomaticTimezone":"false"}
```

Reply:
i will check how to implement the changes you suggest. But the idea of this API is return all timezones for a channel, does not matter the user.

Do you want to change this behavior and return the users timezones in a channel with a user id?

Reply:
No, I was just meaning that instead of returning a list of timezone objects, we just return a list of timezone names like `"Africa/Asmera", "Americas/New York", "GMT"`

Reply:
@hmhealey did the changes, PTAL, if you agree on this I will make the changes in the webapp/redux to support that